ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -828,26 +832,30 @@ func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv
 // (1) shouldDelete: Whether the Pod should be deleted.
 // (2) reason: The reason why the Pod should or should not be deleted.
 func shouldDeletePod(pod corev1.Pod, nodeType rayv1.RayNodeType) (bool, string) {
-	// If a Pod's restart policy is set to `Always`, KubeRay will not delete
+	// If a Pod's restart policy is set to `Always` and Pod not be evicted, KubeRay will not delete
```

Comment:
- and Pod not be evicted -> and the Pod is not evicted
- KubeRay will not delete -> KubeRay will not delete it